shuup_tests/front/test_product_price.py

```diff
@@ -104,3 +198,26 @@ def check(shop_product, input_value, expected_output):
     check(shop_product, '123\0', None)
     check(shop_product, '123\0456', None)
     check(shop_product, '\n', None)
+
+@pytest.mark.parametrize("decimals", [0, 2])
+def test_product_price_get_quantity_with_display_unit(rf, decimals):
+    shop = get_default_shop()
+    product = create_product("sku", shop=shop)
+    shop_product = product.get_shop_instance(shop)
+    from shuup.core.models import SalesUnit, DisplayUnit
+    sales_unit = SalesUnit.objects.create(identifier="random", decimals=decimals)
+    product.sales_unit = sales_unit
+    product.save()
+
+    view = ProductPriceView()
+    view.request = rf.get('/')
+    assert 'quantity' not in view.request.GET
+
+    def check(shop_product, input_value, expected_output):
+        view.request.GET = dict(view.request.GET, quantity=input_value, unitType="not internal")
+        result = view._get_quantity(shop_product)
```

Comment:
shouldnt you do some check here with the expected value?

Reply:
perhaps